Magento_GiftCard/templates/product/view/type/giftcard.phtml

```diff
@@ -136,7 +136,7 @@
 
                             <?= 'data-validate="{maxlength:'
                                 . $block->escapeHtmlAttr($block->getMessageMaxLength())
-                                . ',messages:{maxlength:' . $errorMessage
+                                . ',messages:{maxlength:' . $block->escapeJs($errorMessage)
```

Comment:
I think here we can use `$block->escapeHtmlAttr($block->escapeJs($errorMessage))` but it would have to best tested on storefront.